### DISCUSSION TITLE
Adding a new data model

### DISCUSSION THREAD

**Discussion by hareld21** [alephdata/aleph#1912] [Q&A]
How do I adding a new (custom) data model to Aleph?

**Comment by Rosencrantz** [comment]
Hi @hareld21 

I think probably the best place for you to start with this process is with the documentation. The following link provides some insights on how you might create your own entities for Aleph:

https://followthemoney.readthedocs.io/en/latest/custom.html

Have a read through this and let us know if you have any followup questions.

Kind regards
Steve

**Comment by hareld21** [comment]
Thank you.
Are those entities automatically populated for all documents in the dataset/database uploaded to Aleph?

**Reply by Rosencrantz** [reply]
Not automagically. Generating entities requires working with structured data and then mapping that data to properties within a given entity. 

As long as you have a structured document (think CSV file) that you're importing mapping fields to entities is very straight forward. But, Aleph won't analyse your files and extract entities from them
